pkg/util/httputil/client.go

```diff
@@ -69,6 +75,20 @@ func Post(
 	return DefaultClient.Post(ctx, url, contentType, body)
 }
 
+// Put does like http.Put but uses the provided context and obeys its cancellation.
```

Comment:
Nit: `Put does like http.Put` is awkward; had to pause to disambiguate the meaning. I'd instead say, `Put is like http.Put but...`